### DISCUSSION TITLE
How to use SQL `LAST_INSERT_ID` with $executeRaw/$queryRaw?

### DISCUSSION THREAD

**Discussion by kldzj** [prisma/prisma#3974] [Q&A] [answered]
I'm currently facing a problem where sometimes, due to Prisma choosing a different connection when executing queries, a query for LAST_INSERT_ID() returns 0, even though the $executeRaw statement clearly inserted as requested.

I've also tried directly inserting and querying the last inserted id at the same time, but I think $queryRaw doesn't support multiple statements?

Did anybody try inserting and then querying the inserted rows id using $executeRaw/$queryRaw?

```js
await prisma.$executeRaw`INSERT INTO test (a, b, c) VALUES (${'test'}, ${'test'}, ${'test'})`
const [result] = await prisma.$queryRaw`SELECT LAST_INSERT_ID() AS id`
console.log(result.id) // sometimes logs 0
```

**Comment by ryands17** [comment, marked as answer]
Hey @kldzj 👋 
Could you explain your use case of using a raw query here?

**Reply by kldzj** [reply]
Well, in theory I want to use it in combination with Locks/Transactions, that's why I'm not calling the models CRUD methods.

**Reply by kldzj** [reply]
@ryands17 @janpio I think the query-/executeRaw methods would be way more valuable if Prisma let the developer choose to do two rather simple things:

- Allow force sending non-prepared statements
- Allow getting a (random) connection (or identifier) out of the pool that can be reused by passing it to the various methods

Until now I never had to use the raw API that Prisma exposes, but it turns out it's way more of a hassle than I initially thought.

Anyway, these are just my two cents. I really love using Prisma, please keep up the amazing work!

**Reply by janpio** [reply]
Thanks for the comment, it would be great if you could look for existing issues about this or created new ones (important: separate ones so we can treat this requests by themselves instead of as a package). We love feedback like this and our development is mostly driven by requests like these.

**Reply by kldzj** [reply]
To anyone coming here looking for an actual solution:

Sending non-prepared statements is tracked in #3337 

Connection reusability is (for now) tracked in #3991

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
